src/main/webapp/clusters-view/clusters-view.controller.js

```diff
@@ -12,9 +12,60 @@ angular.module('managementConsole')
             $scope.group = modelController.getServer().getServerGroupName();
             $scope.stack = modelController.getServer().getFirstServer().getDefaultStack();
             $scope.relays = {};
+            $scope.offlineSites = {};
+            $scope.onlineSites  = {};
+            $scope.mixedSites   = {};
+
+            //
+            // Updates the map of remote site status
+            //
+            $scope.refreshRemoteSitesStatus = function(cluster) {
+              var resourcePathCacheContainer = cluster.domain.getFirstServer().getResourcePath()
+                 .concat('subsystem', 'datagrid-infinispan', 'cache-container', cluster.name);
+
+                // Refresh list of offline sites
+                cluster.modelController.readAttribute(resourcePathCacheContainer, 'offline-sites').then(
+                  function (response){
+                      if( response != null && response.constructor === Array) {
+                        $scope.offlineSites[cluster.name] = response
+                      } else {
+                        $scope.offlineSites[cluster.name] = [];
+                      }
+                  }
+                );
+
+                // Refresh list of online sites
+                cluster.modelController.readAttribute(resourcePathCacheContainer, 'online-sites').then(
+                  function (response){
+                      if( response != null && response.constructor === Array) {
+                        $scope.onlineSites[cluster.name] = response
+                      } else {
+                        $scope.onlineSites[cluster.name] = [];
+                      }
+                  }
+                );
+
+                // Refresh list of mixed sites
+                cluster.modelController.readAttribute(resourcePathCacheContainer, 'mixed-sites').then(
+                  function (response){
+                    if( response != null && response.constructor === Array) {
+                      $scope.mixedSites[cluster.name] = response
+                    } else {
+                      $scope.mixedSites[cluster.name] = [];
+                    }
+                  }
+               );
```

Comment:
You can also add catch function to promise chain. I plan to do this throughout our code in the next pass so it is not a must. But see last screenshot on http://terussell85.github.io/blog/2013/12/23/the-angularjs-promise-anti-pattern-that-makes-me-cry/ and how it can be done.